packages/core-ui/lib/modal/index.tsx

```diff
@@ -95,3 +100,144 @@ export const ActionItem = ({
     </h3>
   );
 };
+
+/**
+ * @name ModalPadding
+ * @summary Generic wrapper for modal padding.
+ */
+export const ModalPadding = forwardRef(
+  (
+    { children, style, verticalOnly, horizontalOnly }: ModalPaddingProps,
+    ref?: RefObject<HTMLDivElement>
+  ) => (
+    <div
+      ref={ref}
+      className={`modal-padding${valEmpty(
+        verticalOnly,
+        "vertical-only"
+      )}${valEmpty(horizontalOnly, "horizontal-only")}`}
+      style={style}
+    >
+      {children}
+    </div>
+  )
+);
+ModalPadding.displayName = "ModalPadding";
+
+/**
+ * @name ModalFixedTitle
+ * @summary Fixed the title.
+ */
+export const ModalFixedTitle = forwardRef(
+  (
+    { children, style, withStyle }: ModalFixedTitleProps,
+    ref?: RefObject<HTMLDivElement>
+  ) => (
+    <div
+      ref={ref}
+      className={`modal-fixed-title${valEmpty(withStyle, "with-style")}`}
+      style={style}
+    >
+      {children}
+    </div>
+  )
+);
+ModalFixedTitle.displayName = "ModalFixedTitle";
+
+/**
+ * @name ModalSeparator
+ * @summary A line to separate the content.
+ */
+export const ModalSeparator = () => <div className="modal-separator" />;
+
+/**
+ * @name ModalNotes
+ * @summary Note styling.
+ */
+export const ModalNotes = ({
+  children,
+  style,
+  withPadding,
+}: ModalNotesProps) => (
+  <div
+    className={`modal-notes${valEmpty(withPadding, "with-padding")}`}
+    style={style}
+  >
+    {children}
+  </div>
+);
+
+/**
+ * @name ModalWarnings
+ * @summary Warnings styling.
+ */
+export const ModalWarnings = ({
+  children,
+  style,
+  withMargin,
+}: ModalWarningsProps) => (
+  <div
+    className={`modal-warnings${valEmpty(withMargin, "with-margin")}`}
+    style={style}
+  >
+    {children}
+  </div>
+);
+
+/**
+ * @name ModalCustomHeader
+ * @summary The header section along with the title.
+ */
+export const ModalCustomHeader = ({ children, style }: ComponentBase) => (
+  <div className="modal-custom-header" style={style}>
+    {children}
+  </div>
+);
+
+/**
+ * @name  ModalSection
+ * @summary Section wrapper.
+ */
+export const ModalSection = ({
+  children,
+  style,
+  multiSections,
+  threeSections,
+  tabs,
```

Comment:
Can these props be combined into one `type` prop? `type = 'two-section' | 'three-section' | 'tabs'.

Then `multiSections` will not need to be provided.